pneumonia_functions.py

```diff
@@ -0,0 +1,280 @@
+    input_cfg = os.path.join(project_dir, "darknet/cfg/yolov3.cfg")
+    with open(input_cfg, 'r') as cfg_in:
+        new_cfg = cfg_in.read()
+
+    max_batches = 2000 * class_nbr
+    steps = str(max_batches * 0.8) + ',' + str(max_batches * 0.9)
+    filter_yolo = str((class_nbr + 5) * 3)
+    new_cfg = new_cfg.replace('batch=64', 'batch=' + str(batch))
+    new_cfg = new_cfg.replace('subdivisions=16', 'subdivisions=' + str(subd))
+    new_cfg = new_cfg.replace('max_batches = 500200', 'max_batches =' + str(max_batches))
+    new_cfg = new_cfg.replace('steps=400000,450000', 'steps=' + steps)
+    new_cfg = new_cfg.replace('classes=80', 'classes=' + str(class_nbr))
+    new_cfg = new_cfg.replace('filters=255', 'filters=' + filter_yolo)
+
+    output_cfg = os.path.join(train_data_dir, "yolo-obj.cfg")
+    with open(output_cfg, 'w') as cfg_out:
+        cfg_out.write(new_cfg)
+
+
+def yolo_names_file(train_data_dir, list_names):
+    '''Generate the file gathering all object class names for yolo v3 training
+    We except a list of string and save it under 'obj.names' in data directory'''
+    names_file = os.path.join(train_data_dir, "obj.names")
+    with open(names_file, 'w') as names:
+        for obj_name in list_names:
+            line = "{}\n".format(obj_name)
+            names.write(line)
+
+
+def yolo_data_file(train_data_dir, backup, class_nbr):
+    '''Generate the file with paths for yolo v3 training
+    The file will be save under 'obj.data' in data directory'''
+    data_file = os.path.join(train_data_dir, "obj.data")
+    with open(data_file, 'w') as data:
+        line = f"classes = {class_nbr}\n"\
+        f"train = {train_data_dir + 'train.txt'}\n"\
+        f"valid = {train_data_dir + 'val.txt'}\n"\
+        f"names = {train_data_dir + 'obj.names'}\n"\
+        f"backup = {backup}"
+        data.write(line)
+
+
+def dcm_to_array(image_path):
+    '''Tranform dicom image format to a numpy array'''
+    dcm_im = pydicom.read_file(image_path + ".dcm").pixel_array
+    return np.stack([dcm_im]*3, -1)
+
+
+def yolo_jpg_file(dataset, origin_folder, target_folder):
+    '''Copy the choosen images in the right directory under jpg format'''
+    for image_name in dataset.iloc[:, 0].unique():
+        image = dcm_to_array(os.path.join(origin_folder, image_name))
+        cv2.imwrite(os.path.join(target_folder, image_name + ".jpg"), image)
+
+
+def yolo_label_generation(dataset, target_folder, image_size):
+    '''Generate label in the shape of yolo_v3 learning CNN:
+    <object-class> <x_center> <y_center> <width> <height>
+    relative value is required by yolo_v3 algorithm.
+    one txt file per image is also required by yolo_v3 algorithm
+    As mentionned in darknet repo, to improve results we need to add images without objects
+    All created files will be saved in the same directory which contains jpg files'''
+    for name, groupe in dataset.groupby("patientId"):
```

Comment:
```suggestion
    for name, group in dataset.groupby("patientId"):
```

Reply:
You often confuse english spellings with the french ones. You should check the spellings online as often as possible.

Reply:
Use thesaurus, I use it everyday.
https://www.thesaurus.com/browse/group